hudhook/src/hooks/common.rs

```diff
@@ -0,0 +1,119 @@
+use imgui::*;
+use parking_lot::MutexGuard;
+use windows::Win32::Foundation::{HWND, LPARAM, LRESULT, WPARAM};
+use windows::Win32::UI::Input::KeyboardAndMouse::*;
+use windows::Win32::UI::WindowsAndMessaging::{WHEEL_DELTA, WM_XBUTTONDBLCLK, XBUTTON1, *};
+
+pub(crate) type WndProcType =
+    unsafe extern "system" fn(hwnd: HWND, umsg: u32, wparam: WPARAM, lparam: LPARAM) -> LRESULT;
+
+pub(crate) trait ImguiRendererCommon {
+    fn io_mut(&mut self) -> &mut imgui::Io;
+    fn set_focus(&mut self, focus: bool);
+    fn is_focus(&self) -> bool;
```

Comment:
I see the two implementations for this are basically identical. It'd be great if we could do away with these altogether and find a different semantics for `focus`, but it's ok to leave these in for the time being.

Reply:
Hmm from what I saw, on the standard libraries they usually do `get_something` and `get_something_mut`, maybe I could change the names to that to be more compliant with the usual naming scheme? otherwise, I'm not sure how to proceed.

Reply:
I am guilty of not having followed this myself so far, but the [API guidelines](https://rust-lang.github.io/api-guidelines/checklist.html) suggest to use `name` to get an immutable reference (or a copy, for `Copy` types) and `name_mut` for mutable references. I like this pattern.

(OT: we're making strides here so at some point I want to put out a new release, I want to go over that checklist and refactor everything up to spec)

Reply:
Maybe it would be good to open an issue with such checklist? I'm happy to contribute whenever I can, I've been having a blast using this library tbh :)

Reply:
Great idea! Thank you!